Explorer/Assets/DCL/Multiplayer/Movement/Systems/MultiplayerMovementMessageBus.cs

```diff
@@ -156,22 +109,30 @@ private async UniTaskVoid HandleAsync(SmartWrap<Packet> packet, Participant part
                         isStunned = proto.IsStunned,
                     };
 
-                    Inbox(message, participant.Identity);
+                    Inbox(message, obj.FromWalletId);
                 }
             }
         }
 
         private void Inbox(FullMovementMessage fullMovementMessage, string @for)
         {
-            if (InboxByParticipantMap.TryGetValue(@for, out SimplePriorityQueue<FullMovementMessage>? queue) && !queue.Contains(fullMovementMessage))
-                queue.Enqueue(fullMovementMessage, fullMovementMessage.timestamp);
-            else
-            {
-                var newQueue = new SimplePriorityQueue<FullMovementMessage>(); // TODO (Vit): pooling
-                newQueue.Enqueue(fullMovementMessage, fullMovementMessage.timestamp);
+            QueueFor(@for)?.Enqueue(fullMovementMessage, fullMovementMessage.timestamp);
+        }
 
-                InboxByParticipantMap.Add(@for, newQueue);
+        private SimplePriorityQueue<FullMovementMessage>? QueueFor(string walletId)
+        {
+            if (entityParticipantTable.Has(walletId) == false)
+            {
+                ReportHub.LogWarning(ReportCategory.MULTIPLAYER_MOVEMENT, $"Entity for wallet {walletId} not found");
+                return null;
             }
+
+            var entity = entityParticipantTable.Entity(walletId);
+
+            if (globalWorld.Has<SimplePriorityQueue<FullMovementMessage>>(entity) == false)
+                globalWorld.Add<SimplePriorityQueue<FullMovementMessage>>(entity); // TODO (Vit): pooling
```

Comment:
Can I ask you to introduce pooling here so we can finalize this part?

Reply:
// TODO (Vit): filter out Island messages if Participant is presented in the Room

There also this one, @popuz

Reply:
with the approach that we discussed today, it wont be needed right? because we will manage it on the sender side